src/graphemebreak.c

```diff
@@ -0,0 +1,219 @@
+
+    ch = get_next_char(s, len, &posNext);
+    while (ch != EOS)
+    {
+        enum GraphemeBreakClass b = get_char_gb_class(ch);
+
+        if (a == GBP_E_Base || a == GBP_E_Base_GAZ)
+          rule10Left = true;
+        else if (a != GBP_Extend)
+          rule10Left = false;
+
+        if (a == GBP_Regional_Indicator)
+          EvenRegionalIndicators = !EvenRegionalIndicators;
+        else
+          EvenRegionalIndicators = true;
+
+        if (a == GBP_CR && b == GBP_LF)
+            *brks = GRAPHEMEBREAK_NOBREAK;            // Rule: GB3
+        else if (a == GBP_CR || a == GBP_LF || a == GBP_Control || b == GBP_CR || b == GBP_LF || b == GBP_Control)
+            *brks = GRAPHEMEBREAK_BREAK;              // Rule: GB4 + GB5
+        else if (a == GBP_L && (b == GBP_L || b == GBP_V || b == GBP_LV || b == GBP_LVT))
+            *brks = GRAPHEMEBREAK_NOBREAK;            // Rule: GB6
+        else if ((a == GBP_LV || a == GBP_V) && (b == GBP_V || b == GBP_T))
+            *brks = GRAPHEMEBREAK_NOBREAK;            // Rule: GB7
+        else if ((a == GBP_LVT || a == GBP_T) && (b == GBP_T))
+            *brks = GRAPHEMEBREAK_NOBREAK;            // Rule: GB8
+        else if (b == GBP_Extend || b == GBP_ZWJ)
+            *brks = GRAPHEMEBREAK_NOBREAK;            // Rule: GB9
+        else if (b == GBP_SpacingMark)
+            *brks = GRAPHEMEBREAK_NOBREAK;            // Rule: GB9a
+        else if (a == GBP_Prepend)
+            *brks = GRAPHEMEBREAK_NOBREAK;            // Rule: GB9b
+        else if (rule10Left && b == GBP_E_Modifier)
+            *brks = GRAPHEMEBREAK_NOBREAK;            // Rule: GB10
+        else if (a == GBP_ZWJ && (b == GBP_Glue_After_Zwj || b == GBP_E_Base_GAZ))
+            *brks = GRAPHEMEBREAK_NOBREAK;            // Rule: GB11
+        else if (!EvenRegionalIndicators && (b == GBP_Regional_Indicator))
+            *brks = GRAPHEMEBREAK_NOBREAK;            // Rule: GB12 + GB13
+        else
+            *brks = GRAPHEMEBREAK_BREAK;              // Rule: GB999
+
+        brks++;
+        a = b;
+        ch = get_next_char(s, len, &posNext);
+    }
+}
+
+/**
+ * Sets the grapheme breaking information for a UTF-8 input string.
+ *
+ * @param[in]  s     input UTF-8 string
+ * @param[in]  len   length of the input
+ * @param[in]  lang  language of the input, right now this does not influence the algorithm
+ * @param[out] brks  pointer to the output breaking data, containing
+ *                   #GRAPHEMEBREAK_BREAK or #GRAPHEMEBREAK_NOBREAK.
+ *                   First element in output array is for the break behind the first character
+ *                   the pointer must point to an array with at least as many elements as there
+ *                   are characters in the string
+ */
+void set_graphemebreaks_utf8(
+        const utf8_t *s,
+        size_t len,
+        const char *lang,
+        char *brks)
+{
+    set_graphemebreaks(s, len, brks,
+                   (get_next_char_t)ub_get_next_char_utf8);
```

Comment:
Better to align the left parenthesis with the first argument? More consistent with the rest of the library, and this is also what clang-format can do (well, in my configuration...; should I add it to the repo?).